app/Http/Livewire/OauthClients.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace App\Http\Livewire;
+
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+use Laravel\Passport\Client;
+use Laravel\Passport\ClientRepository;
+use Laravel\Passport\TokenRepository;
+use Livewire\Component;
+
+class OauthClients extends Component
+{
+    public $name;
+    public $redirect;
+    public $editClientId;
+    public $editName;
+    public $editRedirect;
+
+    public $authorizationError;
+
+    protected $clientRepository;
+    protected $tokenRepository;
+
+    public function __construct()
+    {
+        $this->clientRepository = app(ClientRepository::class);
+        $this->tokenRepository = app(TokenRepository::class);
+        parent::__construct();
+    }
```

Comment:
Having a constructor in a livewire component caught my eye. It works but seemed _off_ for livewire.

I found [this thread](https://github.com/livewire/livewire/issues/380) which says you can inject them in "action methods" so would work:

```diff
- public function deleteClient(Client $clientId): void
+ public function deleteClient(ClientRepository $clientRepository, Client $clientId): void
```

I don't think it's necessary to change it but it's something to keep in mind for later.

Reply:
Cool, will look into this.